pages/[...slug].tsx

```diff
@@ -291,27 +282,38 @@ export const GenericPage: React.FC<GenericPageProps> = ({
     <Masthead version={version}>
       <MastheadMenu>
         {desktopNavigation.childrenCollection.items.map(
-          ({ title: parentTitle, path: parentPath, childrenCollection }) => {
+          ({
+            title: parentTitle,
+            path: parentPath,
+            externalUri: parentExternalUri,
+            page: parentPage,
+            childrenCollection: children,
+          }) => {
+            const parentHref = parentPage
+              ? parentPath || '#'
+              : parentExternalUri || '#'
+
             return (
               <MastheadMenuItem
                 key={parentPath}
-                link={<Link href={parentPath}>{parentTitle}</Link>}
+                link={<Link href={parentHref}>{parentTitle}</Link>}
```

Comment:
I wonder if this could be `href={parentHref || '#'}` which reduces the complexity for [setting `parentHref`](https://github.com/Royal-Navy/docs.royalnavy.io/pull/218/files#diff-5e5186614071390e5a8824f814c2a49cd4083adc99ada4220e546a421676ca2aR292-R294).